tools/multimodal/source_format_to_data_juicer_format/wavcaps_to_dj.py

```diff
@@ -0,0 +1,236 @@
+#       'href': '2218-Service-Bell-Help.html',
+#       'caption': 'Someone is ringing a bell.',
+#       'id': '2218',
+#       'duration': 1.5698125,
+#       'audio': 'wav_path',
+#       'download_link': 'http://soundbible.com/grab.php?id=2218&type=wav'
+#   },
+#   ...]
+# }
+#
+# Corresponding Data-Juicer format:
+# {'audios': ['./path/to/audio/2219.flac'],
+#  'text': '<audio>\n'
+#          '[[caption]]: An airplane is landing. <|__dj__eoc|>',
+#  '__dj__meta__': {
+#       'num_captions_per_audio': 1,
+#       'title': 'Airplane Landing Airport',
+#       'description': 'Large commercial airplane landing at an airport runway.',  # noqa: E501
+#       'author': 'Daniel Simion',
+#       'href': '2219-Airplane-Landing-Airport.html',
+#       'caption': 'An airplane is landing.',
+#       'id': '2219',
+#       'duration': 14.1424375,
+#       'audio': 'wav_path',
+#       'download_link': 'http://soundbible.com/grab.php?id=2219&type=wav',
+#       'category': '',
+#       'tags': '' }}
+# {'audios': ['./path/to/audio/2218.flac'],
+#  'text': '<audio>\n'
+#          '[[caption]]: Someone is ringing a bell. <|__dj__eoc|>',
+#  '__dj__meta__': {
+#       'num_captions_per_audio': 1,
+#       'title': 'Service Bell Help',
+#       'description': 'Customer ringing service bell in need of help in a store.',  # noqa: E501
+#       'author': 'Daniel Simion',
+#       'href': '2218-Service-Bell-Help.html',
+#       'caption': 'Someone is ringing a bell.',
+#       'id': '2218',
+#       'duration': 1.5698125,
+#       'audio': 'wav_path',
+#       'download_link': 'http://soundbible.com/grab.php?id=2218&type=wav',
+#       'category': '',
+#       'tags': '' }}
+
+import json
+import os
+from typing import List, Union
+
+import fire
+import jsonlines as jl
+from loguru import logger
+from tqdm import tqdm
+
+from data_juicer.utils.constant import Fields
+from data_juicer.utils.mm_utils import SpecialTokens
+
+
+def creat_meta_filed(num_captions_per_audio, source_meta):
+    meta_dict = {
+        'num_captions_per_audio': num_captions_per_audio,
+        'title': '',
+        'description': '',
+        'author': '',
+        'href': '',
+        'caption': '',
+        'id': '',
+        'duration': '',
+        'audio': '',
+        'download_link': '',
+        'category': '',
+        'tags': ''
+    }
+    for key in source_meta:
+        meta_dict[key] = source_meta[key]
+    return meta_dict
+
+
+def get_all_files(dirname):
+    result = {}
+    for maindir, subdir, file_name_list in os.walk(dirname):
+        for filename in file_name_list:
+            filepath = os.path.join(maindir, filename)
+            result[filename] = filepath
+    return result
+
+
+@logger.catch
+def main(
+    wavcaps_json_path: str,
+    wavcaps_audio_path: str,
+    target_ds_path: str,
+    target_field: Union[str, List[str]] = 'caption',
+    eoc_special_token: str = SpecialTokens.eoc,
+    audio_special_token: str = '<audio>',
```

Comment:
If there is no explicit special token specified by the original dataset, it's recommended to use the default special token from Data-Juicer, which is SpeicalTokens.audio here.

LLaVA use "<image>" in the original dataset to indicate the position of audios, but WavCaps didn't specify such tokens as far as I know.